Docs: Local relative paths not supported by projectService's allowDefaultProject

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

_No response_

### Repro Code

```TypeScript
(see ESLint config)
```


### ESLint Config

```javascript
// @ts-check

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: {
            allowDefaultProject: ['./*.js'],
            defaultProject: 'tsconfig.json'
        },
        tsconfigRootDir: import.meta.dirname,
      },
    },
  },
);
```


### tsconfig

_No response_

### Expected Result

`eslint.config.js` should be picked up by the project service.

### Actual Result

```plaintext
/Users/josh/repos/repros/eslint.config.js
  0:0  error  Parsing error: /Users/josh/repos/repros/eslint.config.js was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject
```

### Additional Info

See repro on: https://github.com/JoshuaKGoldberg/repros/tree/repro-allow-default-project-local-relative-path

Followup to #9674 -> #9675: that work made paths like `'*.js'` work, but not `'./*.js'`.

### Versions

| package                                | version  |
| -------------------------------------- | -------- |
| `@typescript-eslint/eslint-plugin`     | `9.8.0`  |
| `@typescript-eslint/parser`            | `9.8.0`  |
| `@typescript-eslint/typescript-estree` | `9.8.0`  |
| `TypeScript`                           | `5.5.4`  |
| `ESLint`                               | `9.8.0`  |
| `node`                                 | `22.4.1` |

💖

--- a/docs/packages/Parser.mdx
+++ b/docs/packages/Parser.mdx
@@ -337,7 +337,7 @@ module.exports = {
   parser: '@typescript-eslint/parser',
   parserOptions: {
     projectService: {
-      allowDefaultProject: ['./*.js'],
+      allowDefaultProject: ['*.js'],
     },
   },
 };

--- a/packages/typescript-estree/tests/lib/validateDefaultProjectForFilesGlob.test.ts
+++ b/packages/typescript-estree/tests/lib/validateDefaultProjectForFilesGlob.test.ts
@@ -10,7 +10,7 @@ describe('validateDefaultProjectForFilesGlob', () => {
   it('does not throw when options.allowDefaultProject contains a non-** glob', () => {
     expect(() =>
       validateDefaultProjectForFilesGlob({
-        allowDefaultProject: ['./*.js'],
+        allowDefaultProject: ['*.js'],
       }),
     ).not.toThrow();
   });

--- a/packages/website/blog/2024-05-27-announcing-typescript-eslint-v8-beta.mdx
+++ b/packages/website/blog/2024-05-27-announcing-typescript-eslint-v8-beta.mdx
@@ -132,7 +132,7 @@ export default tseslint.config(
         project: ['packages/*/tsconfig.json', 'tsconfig.eslint.json'],
         // Added lines start
         projectService: {
-          allowDefaultProject: ['./*.js'],
+          allowDefaultProject: ['*.js'],
           defaultProject: './tsconfig.json',
         },
         // Added lines end

--- a/packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md
+++ b/packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md
@@ -120,7 +120,7 @@ export default tseslint.config(
         project: ['packages/*/tsconfig.json', 'tsconfig.eslint.json'],
         // Added lines start
         projectService: {
-          allowDefaultProject: ['./*.js'],
+          allowDefaultProject: ['*.js'],
           defaultProject: './tsconfig.json',
         },
         // Added lines end

